driver/esp32/espidf.c

```diff
@@ -171,12 +171,16 @@ typedef struct {
 
 #define DISPLAY_TYPE_ILI9341 1
 #define DISPLAY_TYPE_ILI9488 2
+#define DISPLAY_TYPE_GC9A01  3
+#define DISPLAY_TYPE_ST7789  4
```

Comment:
Is the st7735 exactly the same as st7789?  
PR's title is "st7789 and st7735 Support" but I couldn't find references to the st7735.

Reply:
Enabling start_x and start_y parameters in the ili9341 driver allows it to work with st7735 devices.  Some st7735 devices will work with the st7789 drivers, and others won't. I suspect something in the st7789 init code disagrees with some devices. The st7735's I've tested all work with the modified ili9341 driver. They only require different parameters.

Creating an additional subclass for the st7735 with init code specific to that device might be a better solution. Either way, it should be documented better.